### PR TITLE
Add blacklist agent req support

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Adds the ability to blacklist agent requirements in A6 omnibus
While I was here, I put the path to the agent's static req file as a variable to reuse

### Motivation

A5 actually supported this already
We have the need to not ship certain integrations on certain flavors of linux. For example a dependency of `aerospike` doesn't work with SUSE11. For that reason, we need to not ship the check on SUSE. Just blacklisting the integration isn't enough because the agent's static requirements file is defined in datadog_checks_base. So we need to also have a way to remove the dependency from that static file. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by ofek — DISMISSED** [review]
🥇

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 1]
```diff
@@ -48,8 +48,11 @@
   'ntp',                           # provided as a go check by the core agent
 ]
 
+blacklist_req = []
+
 if suse?
-  blacklist.push('aerospike')  # Temporarily blacklist Aerospike until builder supports new dependency
+  blacklist.push('aerospike')  # Temporarily blacklist Aerospike Datadog Integration until builder supports new dependency
+  blacklist_req.push(/^areospike==/)  # Temporarily blacklist Aerospike python client until builder supports new dependency
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-integrations.rb]: typo: `aerospike`

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 2]
```diff
@@ -167,11 +170,34 @@
       end
     end
 
+    if windows?
+      static_reqs_file = "#{windows_safe_path(install_dir)}\\#{agent_requirements_file}"
+    else
+      static_reqs_file = "#{install_dir}/#{agent_requirements_file}"
+
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-integrations.rb]: missing `end`

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 3]
```diff
@@ -167,11 +170,34 @@
...
+    # Remove any blacklisted requirements from the static-environment req file
+    tmp = []
+    File.open("#{static_reqs_file}", 'r+') do |f|
+      f.each_line do |line|
+        tmp.push(line)
+      end
+    end
+
+    File.open("#{static_reqs_file}", 'w+') do |f|
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-integrations.rb]: I'd use a different file to write the filtered reqs to make things more obvious

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 4]
```diff
@@ -167,11 +170,34 @@
...
     end
 
+    if windows?
+      static_reqs_file = "#{windows_safe_path(install_dir)}\\#{agent_requirements_file}"
+    else
+      static_reqs_file = "#{install_dir}/#{agent_requirements_file}"
+
+    # Remove any blacklisted requirements from the static-environment req file
+    tmp = []
+    File.open("#{static_reqs_file}", 'r+') do |f|
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-integrations.rb]: To read all the lines of the file you can simply use `File.open("#{static_reqs_file}", 'r+').readlines()` here (see https://ruby-doc.org/core-2.1.0/IO.html#method-i-readlines)

**Review by olivielpeau — COMMENTED** [review]
A few comments, overall the logic makes sense 👍

File: omnibus/config/software/datadog-agent-integrations.rb (hunk 3 above)
Comment (nmuesch) [reply omnibus/config/software/datadog-agent-integrations.rb]: I can do that, just have to make sure thats the file we use to `compile` with, which shouldn't be an issue.

File: omnibus/config/software/datadog-agent-integrations.rb (hunk 4 above)
Comment (nmuesch) [reply omnibus/config/software/datadog-agent-integrations.rb]: Nice, was looking for that when working on this PR 😄thanks! As you can tell, I don't often Ruby.

**Review by ofek — APPROVED** [review]

File: omnibus/config/software/datadog-agent-integrations.rb (hunk 4 above)
Comment (truthbk) [reply omnibus/config/software/datadog-agent-integrations.rb]: > I don't often Ruby

Don't say it like it's a bad thing ;)

**Review by truthbk — APPROVED** [review]
Builds look good on gitlab. Thank you @nmuesch! :shipit:

**Pull request merged**